lib/handlers/authorize-handler.js

```diff
@@ -0,0 +1,310 @@
+
+          this.updateResponse(response, redirectUri, state);
+
+          throw e;
+        });
+    });
+};
+
+/**
+ * Generate authorization code.
+ */
+
+AuthorizeHandler.prototype.generateAuthorizationCode = function() {
+  if (this.model.generateAuthorizationCode) {
+    return Promise.try(this.model.generateAuthorizationCode);
+  }
+
+  return tokenUtil.generateRandomToken();
+};
+
+/**
+ * Get authorization code lifetime.
+ */
+
+AuthorizeHandler.prototype.getAuthorizationCodeLifetime = function() {
+  var expires = new Date();
+
+  expires.setSeconds(expires.getSeconds() + this.authorizationCodeLifetime);
+
+  return expires;
+};
+
+/**
+ * Get the client from the model.
+ */
+
+AuthorizeHandler.prototype.getClient = function(request) {
+  var clientId = request.body.client_id || request.query.client_id;
+
+  if (!clientId) {
+    throw new InvalidRequestError('Missing parameter: `client_id`');
+  }
+
+  if (!is.vschar(clientId)) {
+    throw new InvalidRequestError('Invalid parameter: `client_id`');
+  }
+
+  var redirectUri = request.body.redirect_uri || request.query.redirect_uri;
+
+  if (redirectUri && !is.uri(redirectUri)) {
+    throw new InvalidRequestError('Invalid request: `redirect_uri` is not a valid URI');
+  }
+
+  return Promise.try(this.model.getClient, clientId)
+    .then(function(client) {
+      if (!client) {
+        throw new InvalidClientError('Invalid client: client credentials are invalid');
+      }
+
+      if (!client.grants) {
+        throw new InvalidClientError('Invalid client: missing client `grants`');
+      }
+
+      if (!_.contains(client.grants, 'authorization_code')) {
+        throw new UnauthorizedClientError('Unauthorized client: `grant_type` is invalid');
+      }
+
+      if (!client.redirectUri) {
+        throw new InvalidClientError('Invalid client: missing client `redirectUri`');
+      }
+
+      if (redirectUri && client.redirectUri !== redirectUri) {
+        throw new InvalidClientError('Invalid client: `redirect_uri` does not match client value');
+      }
+
+      return client;
+    });
+};
+
+/**
+ * Get scope from the request.
+ */
+
+AuthorizeHandler.prototype.getScope = function(request) {
+  var scope = request.body.scope || request.query.scope;
+
+  if (!is.nqschar(scope)) {
+    throw new InvalidScopeError('Invalid parameter: `scope`');
+  }
+
+  return scope;
+};
+
+/**
+ * Get state from the request.
+ */
+
+AuthorizeHandler.prototype.getState = function(request) {
+  var state = request.body.state || request.query.state;
+
+  if (!state) {
+    throw new InvalidRequestError('Missing parameter: `state`');
+  }
+
+  if (!is.vschar(state)) {
+    throw new InvalidRequestError('Invalid parameter: `state`');
+  }
+
+  return state;
+};
+
+/**
+ * Get user by calling the authenticate middleware.
+ */
+
+AuthorizeHandler.prototype.getUser = function(request, response) {
+  return this.authenticateHandler.handle(request, response).then(function(token) {
```

Comment:
This is an exact problem I had when using your fork. I use on-site sessions to authenticate users I'm granting tokens to, not a pre-existing access token, so it doesn't make sense to require an access_token here (you're passing getUser to the authenticatHandler, which is authenticating the request for OAuth tokens). I forked your fork to add a `getUser` model function similar to what this library already does, but in a hacky way. Ideally there needs to be a more versatile solution than your current one, since you can't expect everyone to pass an access_token body parameter that is already active.